### PR TITLE
Improve CSVInputFormat to set default value to column if no input while deserialization

### PR DESCRIPTION
**Pull request by KevinyhZou**
### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Sometimes use CSVInputFormat  the deserialize txt file, and the input row fields is less than defined, such as we defined a table with 4 column `test_tbl(x String, y String, z String, w String)`, But the input row only have 3 fields, so it will throws exceptions while deserialize, like below
```
ERROR: Line feed found where delimiter (,) is expected. It's like your file has less columns than expected.
And if your file has the right number of columns, maybe it has unescaped quotes in values.
```

So we improve this, if detect the input fields too few, then we can enable a config to set default value to the column

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

File: src/Processors/Formats/Impl/CSVRowInputFormat.cpp [hunk 1]
```diff
@@ -347,6 +347,14 @@ bool CSVFormatReader::checkForSuffix()
...
 }
 
+bool CSVFormatReader::checkForEndOfLine()
+{
+    if (buf->eof() || *buf->position() == '\n' || *buf->position() == '\r')
+        return true;
+    else
+        return false;
+}
+
```
Comment (lgbo-ustc) [inline src/Processors/Formats/Impl/CSVRowInputFormat.cpp:357]:
```c++
bool CSVFormatReader::checkForEndOfLine()
{
    return buf->eof() || *buf->position() == '\n' || *buf->position() == '\r';
}
```

Comment (KevinyhZou) [reply src/Processors/Formats/Impl/CSVRowInputFormat.cpp]: done

**Comment by kardymonds** [comment]
Looks the same https://github.com/ClickHouse/ClickHouse/pull/51273 / https://github.com/ClickHouse/ClickHouse/issues/9671 (input_format_csv_missing_as_default flag)

**Comment by Avogar** [comment]
Let's close it and merge https://github.com/ClickHouse/ClickHouse/pull/51273

**Pull request closed without merging**